kensu/utils/kensu.py

```diff
@@ -375,45 +389,70 @@ def report_with_mapping(self):
-                        except:
-                            from kensu.requests.models import ksu_str
-                            # FIXME weird... should be fine to delete (and try,except too)
-                            if isinstance(stats_df, pd.DataFrame) or isinstance(stats_df, DataFrame) or isinstance(stats_df,Series) or isinstance(stats_df,pd.Series) :
-                                stats = self.extractors.extract_stats(stats_df)
-                            elif isinstance(stats_df, ksu_str):
-                                stats = None
-                            elif isinstance(stats_df, dict):
-                                stats = stats_df
-                            else:
-                                #TODO Support ndarray
-                                stats = None
-                        if stats is not None:
-                            DataStats(pk=DataStatsPK(schema_ref=SchemaRef(by_guid=schema),
-                                                     lineage_run_ref=LineageRunRef(by_guid=lineage_run.to_guid())),
-                                      stats=stats,
-                                      extra_as_json=None)._report()
-                        elif isinstance(stats_df, KensuDatasourceAndSchema):
+                        if isinstance(stats_df, KensuDatasourceAndSchema):
                             stats_df.f_publish_stats(lineage_run.to_guid())
+
+                        else:
+
+                            try:
+                                stats = self.extractors.extract_stats(stats_df)
+                            except:
+                                from kensu.requests.models import ksu_str
+                                # FIXME weird... should be fine to delete (and try,except too)
+                                if isinstance(stats_df, pd.DataFrame) or isinstance(stats_df, DataFrame) or isinstance(stats_df,Series) or isinstance(stats_df,pd.Series) :
+                                    stats = self.extractors.extract_stats(stats_df)
+                                elif isinstance(stats_df, ksu_str):
+                                    stats = None
+                                elif isinstance(stats_df, dict):
+                                    stats = stats_df
+                                else:
+                                    #TODO Support ndarray
+                                    stats = None
+                            if stats is not None and self.compute_delta:
+                                self.schema_stats[schema] = stats
+                                if lineage_run.to_guid() not in self.stats_to_send:
+                                    self.stats_to_send[lineage_run.to_guid()] = {}
+                                self.stats_to_send[lineage_run.to_guid()][schema] = stats
+                                if schema == to_guid:
+                                    if 'nrows' in stats:
```

Comment:
Cool! Although, let’s think if there wouldn’t be another better place to put this